bot/module/tatsu/wrapper.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+
+import aiohttp
+from ratelimit import limits
+
+import bot.module.tatsu.data_structures as ds
+
+
+class ApiWrapper:
+    def __init__(self, key):
+        self.key = key
+        self.base_url = "https://api.tatsu.gg/v1/"
+        self.headers = {"Authorization": key}
+
+    @limits(calls=60, period=60)
+    async def request(self, url):
+        """Directly interact with the API to get the unfiltered results."""
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url=self.base_url + url, headers=self.headers) as result:
+                if result.status != 200:
+                    return result.raise_for_status()
+                return await result.json()
+
+    @limits(calls=60, period=60)
+    async def patch(self, url, payload):
+        async with aiohttp.ClientSession() as session:
+            async with session.patch(
+                url=self.base_url + url, json=payload, headers=self.headers
+            ) as result:
+                if result.status != 200:
+                    return result.raise_for_status()
+                return await result.json()
+
+    async def get_profile(self, user_id: int) -> ds.UserProfile:
+        """Gets a user's profile. Returns a user object on success."""
+        try:
+            result = await self.request(f"users/{user_id}/profile")
+        except Exception as e:
+            raise e
+        try:
+            subscription_renewal = datetime.datetime.strptime(
+                result.get("subscription_renewal"), "%Y-%m-%dT%H:%M:%SZ"
+            )
+        except Exception:
+            subscription_renewal = None
```

Comment:
What kind of exception?

Reply:
If the user does not have a subscription, it will return 'None,' and the `datetime.datetime.strptime` function will raise an error

Reply:
Then it will be good if we write the specific error instead of the general exception like this for clarity e.g. `ValueError` or such